flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/SocketTextStreamFunctionTest.java

```diff
@@ -0,0 +1,237 @@
+            "Sed vitae nisl vitae turpis dapibus lacinia in id elit. Integer lorem dolor, " +
+            "porttitor ut nisi in, tincidunt sodales leo. Aliquam tristique dui sit amet odio bibendum, " +
+            "et rutrum turpis auctor. Morbi sit amet mollis augue, ac rutrum velit. " +
+            "Vestibulum suscipit finibus sapien, et congue enim laoreet consequat.\r\nInteger " +
+            "aliquam metus iaculis risus hendrerit maximus. Suspendisse vestibulum nibh ac " +
+            "mauris cursus molestie sit amet vel turpis. Nulla et posuere orci. " +
+            "Aliquam dui quam, posuere vitae erat vitae, finibus commodo ipsum. Aliquam " +
+            "eu dui quis arcu porttitor sollicitudin. Integer sodales finibus ullamcorper. " +
+            "Praesent et felis tempor, laoreet libero eget, consequat nisl. " +
+            "Aenean molestie rutrum lorem, ac cursus nisl dapibus vitae.\r\nQuisque sodales " +
+            "dui et sem bibendum semper. Pellentesque luctus leo nec lacus euismod pellentesque. " +
+            "Phasellus a metus dignissim risus auctor lacinia. Class " +
+            "aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos" +
+            ". Aenean consectetur bibendum imperdiet. Etiam dignissim rutrum enim, " +
+            "non volutpat nisi condimentum sed. Quisque condimentum ultrices est sit amet " +
+            "facilisis.\r\nUt vitae volutpat odio. Sed eget vestibulum libero, eu " +
+            "tincidunt lorem. Nam pretium nulla nisl. Maecenas fringilla nunc ut turpis consectetur, " +
+            "et fringilla sem placerat. Etiam nec scelerisque nisi, at sodales ligula. Aliquam " +
+            "euismod faucibus egestas. Curabitur eget enim quam. Praesent convallis mattis lobortis. " +
+            "Pellentesque a consectetur nisl. Duis molestie diam est. Nam a malesuada augue. " +
+            "Vivamus enim massa, luctus ac elit ut, vestibulum laoreet nulla. " +
+            "Curabitur pellentesque vel mi eget tempus. Donec cursus et leo quis viverra.\r\nIn ac imperdiet ex, " +
+            "nec aliquet erat. Nullam sit amet enim in dolor finibus convallis id eu nibh. Fusce aliquam convallis orci aliquam.";
+    private static final byte[] data = content.getBytes();
+
+    @Test
+    public void testNewLineDelimitedOldApiWithChar() throws Exception {
+        List<String> actualList = new ArrayList<>();
+
+        Socket socket = mock(Socket.class);
+        when(socket.getInputStream()).thenReturn(new ByteArrayInputStream(data));
+        when(socket.isClosed()).thenReturn(false);
+        when(socket.isConnected()).thenReturn(true);
+
+        SocketTextStreamFunction source = new SocketTextStreamFunction("", 0, '\n', 0);
+        Field field = SocketTextStreamFunction.class.getDeclaredField("isRunning");
+        field.setAccessible(true);
+        field.set(source, true);
```

Comment:
I do not think it is good to use reflection since I did something the same as you before and they told don`t do this.
Why not set up a socket server in your test?